### ISSUE TITLE
Carla Canvas Patchbay don't save positions

### ISSUE THREAD

**Issue by drachehavoc** [falkTX/Carla#1666]
I'm organizing my canvas and saving a project, but when I close Carla and reopen all the elements are repositioned in random:

I organized in this positions:
![image](https://user-images.githubusercontent.com/1902294/186725442-a3aca9f7-c1aa-4ec4-9b87-83ee730dd108.png)

 
but after closing and reopening carla, they are in these positions:
![image](https://user-images.githubusercontent.com/1902294/186725511-3b2c07d8-47a6-4d11-b96f-58fe51b9d351.png)

I'm using Fedora 36 (Desktop)
![image](https://user-images.githubusercontent.com/1902294/186726044-30b0d60d-66e1-465a-bfe1-a734ec760ae0.png)

**Comment by mxmilkiib** [comment]
What version of JACK are you using?

**Comment by BlueMax** [comment]
Does it help/change if you reload the project (several times)?
Might be the known bug. There are some threads already. I have it by a 50% chance as well.

**Comment by drachehavoc** [comment]
> What version of JACK are you using?

I'm using pipewire and the Carla from flathub, but I already tested the version installed with dnf but the issue stills.

**Comment by drachehavoc** [comment]
> Does it help/change if you reload the project (several times)? Might be the known bug. There are some threads already. I have it by a 50% chance as well. I'm on Pipewire but that's unrelated.

the only error i get is ```Gtk-Message: 20:51:51.964: Failed to load module "canberra-gtk-module"```.

![image](https://user-images.githubusercontent.com/1902294/187051799-e877b900-7241-4c64-b6ff-43b96ccedf83.png)

**Comment by hellocatfood** [comment]
FYI I have this happen sporadically and I'm using `5:2.5.0+git20220726` from the kxstudio repositories, so it's not just the flathub version

**Comment by drachehavoc** [comment]
up, can anyone help with it?

**Comment by BlueMax** [comment]
You haven't answered my question. Do the nodes fix themselves when you reload the project? 'Monitor' nodes do not count. They have to be fixed with [this](https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-JACK#jackmerge-monitor).

**Comment by drachehavoc** [comment]
> You haven't answered my question. Do the nodes fix themselves (reposition) when you reload the project (restart Carla)? 
No, they make the right connections, but they don't go to the position I saved them from.

> 'Monitor' nodes (Pipewire specific) do not count. They have to be fixed with [this](https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-JACK#jackmerge-monitor) first.
I don`t understand how it affect the Carla sources positions, sorry. do you me to change ```jack.merge-monitor```  to make some test?

I'm sorry for not understanding. can you be more didactic?

**Comment by BlueMax** [comment]
There might be a race condition (bug) or something in Carla that leads to nodes not getting positioned properly on a project load. It happens now and then, especially when Carla is freshly started. If this happens you can simply reload the project and nodes are magically re-positioned properly. If that's not the case for you you might have another bug.

Then there's another Pipewire-only issue regarding the 'Monitor' nodes. These are the hardware 'input' nodes that end with 'Monitor' in its node name. You have two of them on your screenshot. These nodes generally don't get properly restored in Carla until you set 'jack.merge-monitor' to 'true'. You can do this e.g. by creating this and restarting Pipewire:

~/.config/pipewire/jack.conf.d/01-jack-monitor.conf
```
jack.properties = {
     jack.merge-monitor = true
}
```
You can tell its in effect when the nodes have 'Monitor' omitted in the name (and get saved properly in Carla).

**Comment by sfjuocekr** [comment]
> There might be a race condition (bug) or something in Carla that leads to nodes not getting positioned properly on a project load. It might happen now and then, especially when Carla is freshly started. If this happens you can simply reload the project and nodes should magically re-position properly. If that's not the case for you you might have another bug.
> 
> Then there's another Pipewire-only issue regarding the 'Monitor' nodes. These are the hardware 'input' nodes that end with 'Monitor' in its node name. You have two of them on your screenshot. These nodes generally don't get properly restored in Carla until you set 'jack.merge-monitor' to 'true'. You can do this e.g. by creating this and restarting Pipewire:
> 
> ~/.config/pipewire/jack.conf.d/01-jack-monitor.conf
> 
> ```
> jack.properties = {
>      jack.merge-monitor = true
> }
> ```
> 
> You can tell its in effect when the nodes have 'Monitor' omitted in the name (and get saved properly in Carla).

I've never faced this, I have always used  jack.merge-monitor = true and today after I pressed CTRL+S to save (which I never do, I just load one preset) positions aren't stored anymore.

Normally I would save the project and it would store the positions, but that is not the case now on 2.5.8. I was on 2.5.7 before, that is when I saved my "mixer" project last.

**Comment by yuceltoluyag** [comment]
I struggled with this issue for days and tried many different methods to fix it. Eventually, I couldn't take it anymore and posted my problem in the Linux audio group. Since no one responded, I thought I was the only one facing it. But it turns out others were experiencing the same thing, and that made me feel a bit relieved :) #1687  https://www.reddit.com/r/linuxaudio/comments/1nnhxoj/carla_doesnt_save_my_connections/
